cmd/oracled/cmd/get_oracle_key.go

```diff
@@ -28,18 +34,41 @@ func getOracleKeyCmd() *cobra.Command {
 
 			uniqueID := svc.EnclaveInfo().UniqueIDHex()
 			oracleAddress := svc.OracleAcc().GetAddress()
-			oracleRegistration, err := svc.QueryClient().GetOracleRegistration(ctx, uniqueID, oracleAddress)
+
+			from, err := cmd.Flags().GetString(flags.FlagFromOracleRegistrationOrUpgrade)
 			if err != nil {
-				return fmt.Errorf("failed to get oracle registration: %w", err)
+				return err
 			}
 
-			if len(oracleRegistration.EncryptedOraclePrivKey) == 0 {
-				return fmt.Errorf("the encrypted oracle private key has not set yet. please try again later")
-			}
+			switch from {
+			case fromRegistration:
+				oracleRegistration, err := svc.QueryClient().GetOracleRegistration(ctx, uniqueID, oracleAddress)
+				if err != nil {
+					return fmt.Errorf("failed to get oracle registration: %w", err)
+				}
 
-			return key.RetrieveAndStoreOraclePrivKey(ctx, svc, oracleRegistration.EncryptedOraclePrivKey)
+				if len(oracleRegistration.EncryptedOraclePrivKey) == 0 {
+					return fmt.Errorf("the encrypted oracle private key has not set yet. please try again later")
+				}
+				return key.RetrieveAndStoreOraclePrivKey(ctx, svc, oracleRegistration.EncryptedOraclePrivKey)
+
+			case fromUpgrade:
+				oracleUpgrade, err := svc.QueryClient().GetOracleUpgrade(ctx, uniqueID, oracleAddress)
+				if err != nil {
+					return fmt.Errorf("failed to get oracle upgrade: %w", err)
+				}
+				if len(oracleUpgrade.EncryptedOraclePrivKey) == 0 {
+					return fmt.Errorf("the encrypted oracle private key has not set yet. please try again later")
+				}
+				return key.RetrieveAndStoreOraclePrivKey(ctx, svc, oracleUpgrade.EncryptedOraclePrivKey)
+
+			default:
+				return fmt.Errorf("invalid --from flag input. please put \"registration\" or \"upgrade\"")
+			}
 		},
 	}
 
+	cmd.Flags().String(flags.FlagFromOracleRegistrationOrUpgrade, "upgrade", "where to get the key from")
```

Comment:
Is `upgrade` as the default value intended?
Then I think you can change like below:
```suggestion
	cmd.Flags().String(flags.FlagFromOracleRegistrationOrUpgrade, fromUpgrade, "where to get the key from")
```

Reply:
I missed it. Thank you.